components/motor/gpio/motor_encoder.go

```diff
@@ -256,22 +257,49 @@ func (m *EncodedMotor) goForMath(ctx context.Context, rpm, revolutions float64)
 // If revolutions is 0, this will run the motor at rpm indefinitely
 // If revolutions != 0, this will block until the number of revolutions has been completed or another operation comes in.
 func (m *EncodedMotor) GoFor(ctx context.Context, rpm, revolutions float64, extra map[string]interface{}) error {
-	m.opMgr.CancelRunning(ctx)
 	ctx, done := m.opMgr.New(ctx)
 	defer done()
 
 	goalPos, goalRPM, direction := m.goForMath(ctx, rpm, revolutions)
 
-	return m.goForInternal(ctx, goalRPM, goalPos, direction)
+	m.goForInternal(goalRPM, goalPos, direction)
+
+	// return and run the motor at rpm indefinitely
+	if revolutions == 0 {
+		return nil
+	}
+
+	positionReached := func(ctx context.Context) (bool, error) {
+		var errs error
+		pos, posErr := m.position(ctx, extra)
+		errs = multierr.Combine(errs, posErr)
+		if (direction == 1 && pos >= goalPos) || (direction == -1 && pos <= goalPos) {
+			stopErr := m.Stop(ctx, extra)
+			errs = multierr.Combine(errs, stopErr)
+			return true, errs
+		}
+		return false, errs
+	}
+	err := m.opMgr.WaitForSuccess(
+		ctx,
+		10*time.Millisecond,
+		positionReached,
+	)
+	// Ignore the context canceled error - this occurs when the rpmCtx is canceled
+	// with m.rpmMonitorDone in goForInternal and in Stop
+	if !errors.Is(err, context.Canceled) {
+		return err
+	}
+	return nil
 }
 
-func (m *EncodedMotor) goForInternal(ctx context.Context, rpm, goalPos, direction float64) error {
+func (m *EncodedMotor) goForInternal(rpm, goalPos, direction float64) {
 	switch speed := math.Abs(rpm); {
 	case speed < 0.1:
-		m.logger.CWarn(ctx, "motor speed is nearly 0 rev_per_min")
-		return motor.NewZeroRPMError()
+		m.logger.Error(motor.NewZeroRPMError())
+		return
```

Comment:
Oof, when I suggested getting rid of the return value, I totally missed that it was actually used here! I'm sorry about that. We should probably undo that last suggestion and go back to returning the error here.

Reply:
honestly don't know why we were logging a warning and returning an error